src/gdb_main.c

```diff
@@ -41,7 +41,7 @@
 #include "rtt.h"
 #endif
 
-#if ADVERTISE_NOACKMODE == 1
+#if defined(ADVERTISE_NOACKMODE) && ADVERTISE_NOACKMODE == 1
```

Comment:
We should probably pass `-DADVERTISE_NOACKMODE=0` in the build system rather than do this - that'll make it more directly comparable to what we're doing in the Meson build system to land, and more consistent to the other enable macros.